app/src/test/java/com/quran/labs/androidquran/model/translation/ArabicDatabaseUtilsTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.quran.labs.androidquran.model.translation
+
+import android.content.Context
+
+import com.quran.data.core.QuranInfo
+import com.quran.data.model.bookmark.Bookmark
+import com.quran.data.pageinfo.common.MadaniDataSource
+import com.quran.labs.androidquran.database.DatabaseHandler
+import com.quran.labs.androidquran.util.QuranFileUtils
+
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import org.mockito.MockitoAnnotations
+import com.google.common.truth.Truth.assertThat
+
+import java.util.ArrayList
+import java.util.HashMap
+
+class ArabicDatabaseUtilsTest {
+
+  @Mock
+  lateinit var context: Context
+
+  @Mock
+  lateinit var arabicHandler: DatabaseHandler
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.openMocks(this@ArabicDatabaseUtilsTest)
+  }
+
+  @Test
+  fun testHydrateAyahText() {
+    val arabicDatabaseUtils = getArabicDatabaseUtils()
+
+    val bookmarks = ArrayList<Bookmark>(3)
+    bookmarks.add(Bookmark(1, 1, 1, 1))
+    bookmarks.add(Bookmark(2, null, null, 3))
+    bookmarks.add(Bookmark(3, 114, 6, 604))
+
+    val result = arabicDatabaseUtils.hydrateAyahText(bookmarks)
+    assertThat(result).hasSize(3)
+
+    assertThat(result[0].ayahText).isNotEmpty()
+    assertThat(result[1].ayahText).isNull()
+
+    assertThat(result).isNotSameInstanceAs(bookmarks)
+  }
+
+  @Test
+  fun testHydrateAyahTextEmpty() {
+    val arabicDatabaseUtils = getArabicDatabaseUtils()
+
+    val bookmarks = ArrayList<Bookmark>(1)
+    bookmarks.add(Bookmark(1, null, null, 3))
+
+    val result = arabicDatabaseUtils.hydrateAyahText(bookmarks)
+    assertThat(result).hasSize(1)
+    assertThat(result[0].ayahText).isNull()
+    assertThat(result).isSameInstanceAs(bookmarks)
+  }
+
+  private fun getArabicDatabaseUtils(): ArabicDatabaseUtils {
+    return object : ArabicDatabaseUtils(context,
+      QuranInfo(MadaniDataSource()),
+      mock(QuranFileUtils::class.java)) {
+
+      override fun getArabicDatabaseHandler(): DatabaseHandler {
+        return arabicHandler
+      }
+
+      override fun getAyahTextForAyat(ayat: List<Int>): Map<Int, String> {
+        val result: MutableMap<Int, String> = HashMap()
+        for (ayahId in ayat) {
+          result[ayahId] = "verse $ayahId"
+        }
+        return result
```

Comment:
how about

```kotlin
return ayat.map { it to "verse $it" }.toMap()
```

Reply:
may you also do this one @wiryadev, then ready to merge - jazakumAllah khairan!

Reply:
sure i can, but i still use `ayahId`  instead of `it` for readability purpose, so it would be like this. Is it okay?
```
return ayat.map { ayahId -> ayahId to "verse $ayahId" }.toMap()
```